lib/std/array_hash_map.zig

```diff
@@ -605,7 +605,11 @@ pub fn ArrayHashMapUnmanaged(
 
         const Self = @This();
 
-        const linear_scan_max = 8;
+        const linear_scan_max = @as(comptime_int, @max(1, @as(comptime_int, @min(
+            std.atomic.cache_line / @as(comptime_int, @max(1, @sizeOf(Hash))),
+            std.atomic.cache_line / @as(comptime_int, @max(1, @sizeOf(K))),
+            std.atomic.cache_line / @as(comptime_int, @max(1, @sizeOf(V))),
+        ))));
```

Comment:
Why not? It seems to produce identical result
```suggestion
        const linear_scan_max: comptime_int = @max(1, @min(
            std.atomic.cache_line / @max(1, @sizeOf(Hash)),
            std.atomic.cache_line / @max(1, @sizeOf(K)),
            std.atomic.cache_line / @max(1, @sizeOf(V)),
        ));
```

Reply:
compile error because `@max` and `@min` (incorrectly) return a sized integer even when the value is comptime-known

Reply:
`comptime_int` is an awkward type which is fundamentally at odds with the range refinement we want `@min`/`@max` to do -- our saviour will hopefully be #3806

Reply:
I disagree; comptime_int is basically a ranged integer with the min and max set to the same value. Strongly believe that `@max` and `@min` should return comptime_int when the value is comptime known given that #3806 is not implemented

Reply:
That would lead to a major behavioral inconsistency between runtime and comptime.